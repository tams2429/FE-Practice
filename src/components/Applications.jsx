import React, {useEffect, useState} from "react";
import axios from "axios";

//COMPONENTS
import SingleApplication from "./SingleApplication";
import { Button } from "../ui/Button/Button.jsx";

//STYLES
import styles from "../css/Applications.module.css";

const Applications = () => {
  
  const [data, setData] = useState([]);
  const [totalData, setTotalData] = useState([]);
  const [loadMore, setLoadMore] = useState(true);
  const [limitPerPage, setLimitPerPage] = useState(5);
  const [currentPagination, setCurrentPagination] = useState(1);

  useEffect(() => {
    const getApplicationTotalData = async () => {
      try {
        const { data } = await axios.get("http://localhost:3001/api/applications");
        return setTotalData(data)
      } catch (error) {
        console.log(error);
      }
    }
    getApplicationTotalData()
  }, [])

  useEffect(() => {
    const getApplicationData = async (currentPagination, limitPerPage) => {
      try {
        const { data } = await axios.get(
          `http://localhost:3001/api/applications?_page=${currentPagination}&_limit=${limitPerPage}`
        );
        return setData(data)
      } catch (error) {
        console.log(error);
      }
    }
    getApplicationData(currentPagination, limitPerPage)

  }, [currentPagination, limitPerPage])

  const handleClick = () => {
    if ((currentPagination + 1) >= (totalData.length) / limitPerPage) {
      setLoadMore(false)
    }
    return setCurrentPagination(currentPagination + 1)
  }

  return (
    <div className={styles.Applications}>
      {data && data.map((user, index) => {
        return <SingleApplication key={index} application={user} />
      })}
      <Button onClick={handleClick} disabled={!loadMore}>Load more</ Button>
    </div>
  );
};

export default Applications;
