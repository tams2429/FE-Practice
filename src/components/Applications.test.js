import {
  render,
  screen,
  fireEvent,
  waitFor,
  findByText,
} from "@testing-library/react";
import Applications from "./Applications";
import axios from "axios";

jest.mock("axios");
const fakeUsers = [
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 37597,
    first_name: "Miles",
    last_name: "Espinoza",
    company: "Qnekt",
    email: "milesespinoza@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 37597,
    first_name: "Miles",
    last_name: "Espinoza",
    company: "Qnekt",
    email: "milesespinoza@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 37597,
    first_name: "Miles",
    last_name: "Espinoza",
    company: "Qnekt",
    email: "milesespinoza@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 37597,
    first_name: "Miles",
    last_name: "Espinoza",
    company: "Qnekt",
    email: "milesespinoza@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 37597,
    first_name: "Miles",
    last_name: "Espinoza",
    company: "Qnekt",
    email: "milesespinoza@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
  {
    guid: "8a8f6cbc-77a1-4086-8968-a57816f4ff60",
    loan_amount: 123,
    first_name: "John",
    last_name: "Smith",
    company: "Qnekt",
    email: "johnsmith@qnekt.com",
    date_created: "2021-08-10",
    expiry_date: "2021-12-02",
  },
];

describe("Application Component", () => {
  let applicationComponent;

  it("should match snapshot", async () => {
    axios.get.mockResolvedValue({ data: fakeUsers });
    applicationComponent = render(<Applications />);
    await waitFor(() => {
      expect(screen.getAllByText("Company")).not.toBeNull();
    });
    expect(applicationComponent.container).toMatchSnapshot();
  });
  it("should call the api endpoint on first render", () => {
    //To be written with more time
  });
  it("should call api endpoint to fetch more data when the 'Load more' button is clicked", () => {
    //To be written with more time
  });

  it("should disable button when the end of the data is reached", () => {
    //To be written with more time
  });
});
