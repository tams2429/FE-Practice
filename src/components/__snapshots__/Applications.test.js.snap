// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Component should match snapshot 1`] = `
<div>
  <div
    class="Applications"
  >
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        Miles
         
        Espinoza
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        milesespinoza@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        37597
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        Miles
         
        Espinoza
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        milesespinoza@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        37597
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        Miles
         
        Espinoza
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        milesespinoza@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        37597
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        Miles
         
        Espinoza
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        milesespinoza@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        37597
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        Miles
         
        Espinoza
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        milesespinoza@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        37597
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <div
      class="SingleApplication"
    >
      <div
        class="cell"
      >
        <sub>
          Company
        </sub>
        Qnekt
      </div>
      <div
        class="cell"
      >
        <sub>
          Name
        </sub>
        John
         
        Smith
      </div>
      <div
        class="cell"
      >
        <sub>
          Email
        </sub>
        johnsmith@qnekt.com
      </div>
      <div
        class="cell"
      >
        <sub>
          Loan Amount
        </sub>
        £
        123
      </div>
      <div
        class="cell"
      >
        <sub>
          Application Date
        </sub>
        10th August 2021
      </div>
      <div
        class="cell"
      >
        <sub>
          Expiry date
        </sub>
        2nd December 2021
      </div>
    </div>
    <button
      class="button"
    >
      Load more
    </button>
  </div>
</div>
`;
